### ISSUE TITLE
Firefox Compatibility Bugs

### ISSUE THREAD

**Issue by pmarks-net** [pmarks-net/ipvfoo#32]
I'm creating this Issue to track problems that occur only when running [IPvFoo on Firefox](https://addons.mozilla.org/en-us/firefox/addon/ipvfoo-pmarks/).

- Cached requests do not include the IP address.
  reported at https://bugzilla.mozilla.org/show_bug.cgi?id=1395020
- Popup window has a horizontal scrollbar.
  reported at https://bugzilla.mozilla.org/show_bug.cgi?id=1395025
- When I right-click an IP address, the "Look up address on bgp.he.net" menu has the wrong enabled/disabled state.
  reported at https://bugzilla.mozilla.org/show_bug.cgi?id=1439497

**Comment by Gitoffthelawn** [comment]
For those following this issue, a fork with updates has been released:

https://github.com/ondrejvrabel/ipvfoobar  
https://addons.mozilla.org/firefox/addon/ipvfoobar/  

It's possible that Firefox compatibility has been improved with this fork (I haven't had time to go through the source code and then try this fork).

**Comment by tdulcet** [comment]
> It's possible that Firefox compatibility has been improved with this fork

That fork does not improve Firefox compatibility unfortunately, it just replaces the context menu with links directly in the popup.

I created a similar add-on called "Server Status" for Firefox only: https://github.com/tdulcet/Server-Status
![requests](https://user-images.githubusercontent.com/15305150/181918341-8fa6fab2-7324-4116-98f8-8da27a2bd7c3.png)
Like IPvFoo, it shows the IP address for all the requests on the page and supports both IPv4 and IPv6. However, it shows all the IP addresses used for each hostname, not just the last one, it highlights the suffix for the hostnames using Mozilla's [Public Suffix List](https://publicsuffix.org/) and it shows the location of each IP address using an IP geolocation database.

It also shows much more information for each hostname, including the number of connections, the days left until certificate expiration, the SSL/TLS version, the HSTS status and the HTTP status code(s). While it obviously is not able to fix those two Firefox bugs mentioned in this issue that have not yet been fixed, it was built specifically to support Firefox and is able to make use of several APIs and features not available in Chrome. As with IPvFoo, all the information is determined locally in your browser to respect your privacy.

It is currently available for download from AMO: https://addons.mozilla.org/firefox/addon/server-status/. Feedback is welcome.

**Comment by Gitoffthelawn** [comment]
Thanks Teal.  That looks very promising.

**Comment by pmarks-net** [comment]
I think I fixed everything:

- **Cached requests do not include the IP address:** Added `ipCache` to hold recent IP addresses in RAM, which fixes the most egregious `(no address)` situations.

- **Popup window has a horizontal scrollbar:** Added `scrollbarHack()` with random bullshit that seems to work.

- **"Look up address on bgp.he.net" menu has the wrong enabled/disabled state:** This broke on Chrome also, so years ago I replaced the disabled state with a shake animation.
